android-components/components/lib/crash-sentry/src/test/java/mozilla/components/lib/crash/sentry/SentryServiceTest.kt

```diff
@@ -263,4 +269,48 @@ class SentryServiceTest {
         verify(service).prepareReport(breadcrumbs, SentryLevel.INFO)
         verify(service).reportToSentry(exception)
     }
+
+    @Test
+    @Suppress("UnstableApiUsage")
```

Comment:
❓ JOOC is there a way to not add this, I see it's used for `SentryExceptionFactory`. 

Could we hook into the sent events and just assert that the sent event had the expected mechanism? That way it would be more of and input/output test and we won't have to call `execute` on `beforeSendCallback`, we could init Sentry and report the event which would trigger the beforeSendCallback like in production.

Reply:
@rahulsainani spent some time digging through the library and I'm not sure there would be a good way to test the sentry library in its entirety. I think I did find a better way to implement both this and the rust crash modifier though!